src/BenchmarkDotNet/Validators/DotNetSdkProvider.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace BenchmarkDotNet.Validators
+{
+    public class DotNetSdkProvider : ISdkProvider
+    {
+        private string _customDotNetCliPath;
+
+        public string CustomDotNetCliPath
+        {
+            get => _customDotNetCliPath;
+            set => _customDotNetCliPath = value;
+        }
```

Comment:
Just make this a read only property and pass the string into the constructor (can make it an optional argument).